src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -5572,9 +6265,8 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
                 break;
             }
             case ASR::ttypeType::Struct: {
-                ASR::Struct_t* der = ASR::down_cast<ASR::Struct_t>(x->m_type);
-                ASR::StructType_t* der_type = ASR::down_cast<ASR::StructType_t>(
-                    ASRUtils::symbol_get_past_external(der->m_derived_type));
+                ASR::Struct_t* der = (ASR::Struct_t*)(&(x->m_type->base));
+                ASR::StructType_t* der_type = (ASR::StructType_t*)(&(der->m_derived_type->base));
```

Comment:
To be reverted, I think. 
https://github.com/lcompilers/lpython/pull/1579/files#diff-48a7161397f9eb8b730f0cbec2f11ce1471aac961dc6b16dc558cf161c375054R5566